compiler/luci-interpreter/src/kernels/FloorDiv.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ * Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kernels/FloorDiv.h"
+#include "kernels/Utils.h"
+
+#include <tensorflow/lite/kernels/internal/reference/binary_function.h>
+#include <math.h>
+
+namespace luci_interpreter
+{
+
+namespace kernels
+{
+
+FloorDiv::FloorDiv(const Tensor *input, const Tensor *alpha, Tensor *output)
+    : Kernel({input, alpha}, {output})
+{
+}
+
+void FloorDiv::configure()
+{
+  LUCI_INTERPRETER_CHECK(x()->element_type() == output()->element_type());
+  LUCI_INTERPRETER_CHECK(y()->element_type() == output()->element_type());
+
+  output()->resize(calculateShapeForBroadcast(x()->shape(), y()->shape()));
+}
+
+void FloorDiv::execute() const
+{
+  switch (x()->element_type())
+  {
+    case DataType::FLOAT32:
+      evalFloat();
+      break;
+    default:
+      throw std::runtime_error("Unsupported type.");
+  }
+}
+
+void FloorDiv::evalFloat() const
+{
+  auto FloorDivFunc = [](float x, float y) -> float {
+    return std::floor(std::divides<double>()(static_cast<double>(x), static_cast<double>(y)));
+  };
+
```

Comment:
Please check denominator (`y`) is non-zero before calling TFLite kernels.

It would be even better to add a test for a divide-by-zero case.

Reply:
Isn't division-by-zero a defined behavior for IEEE 754 floats? I've tested it with constant tensors and it gives valid output with `inf`/`-inf` values (or `nan` in case of `0/0`).

I've updated test cases to cover this situations and specific case for NaN. Please correct me if I understood it wrong.

Reply:
TF tensors can have inf/-inf/nan (https://www.tensorflow.org/api_docs/python/tf/math/is_inf), so it makes sense to support div-by-zero.

But, I saw the check code from the TFLite kernel. If we allow div-by-zero, luci-interpreter may work differently from TFLite interpreter. I'm not sure this is desirable. @s-barannikov what do you think of this?

Reply:
Haven't noticed this check in tflite, thanks! It seems that tflite doesn't support zero-denominators then, I've added check and `_NEG` testcase for it.